sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/PagedConverter.java

```diff
@@ -28,36 +29,58 @@ private PagedConverter() {
     /**
      * Applies flatMap transform to elements of PagedFlux.
      *
-     * @param <T> input type of pagedFlux
-     * @param <S> return type of pagedFlux
-     * @param pagedFlux input
-     * @param mapper the flatMap transform of element T to Publisher of S.
-     * @return the PagedFlux.
+     * @param pagedFlux the input of PagedFlux.
+     * @param transformer the flatMap transform of element T to Publisher of S.
+     * @param <T> input type of PagedFlux.
+     * @param <S> return type of PagedFlux.
+     * @return the PagedFlux with elements in PagedResponse transformed.
      */
     public static <T, S> PagedFlux<S> flatMapPage(PagedFlux<T> pagedFlux,
-            Function<? super T, ? extends Publisher<? extends S>> mapper) {
+            Function<? super T, ? extends Publisher<? extends S>> transformer) {
         Supplier<PageRetriever<String, PagedResponse<S>>> provider = () -> (continuationToken, pageSize) -> {
             Flux<PagedResponse<T>> flux = (continuationToken == null)
                     ? pagedFlux.byPage()
                     : pagedFlux.byPage(continuationToken);
-            return flux.flatMap(PagedConverter.flatMapPagedResponse(mapper));
+            return flux.concatMap(PagedConverter.flatMapPagedResponse(transformer));
```

Comment:
the name of `flatMapPage` is actually about the `Function<? super T, ? extends Publisher<? extends S>> mapper` transform function. It is map of one to Flux, hence `flatMap`.

It is not about the `concatMap` on parent `PagedFlux`, it will always be `concatMap`, process one page, then next page (lazy fetch by page, allow user can cancel in between).

Reply:
`flatMap` and `concatMap` could take same `mapper` to map `flux`, the only different is that `flatMap` do it as soon as possible, but the `concatMap` will do it one after the previous one finished.

If the `flatMap` and the `concatMap` behave same in `PagedFlux`, then it could be OK.

Reply:
I think we want `concatMap` here, point here to do one page after another, so we do not eagerly fetch results (and apply the mapper) on next page.

Within page, it uses `flatMapSequential` to fetch is eagerly, while still keep the item order.